0000-pre-shared-key-v2.md

```diff
@@ -0,0 +1,291 @@
+
+### Encryption
+
+In PSKv2.0, request and response data is encrypted using [AES-GCM](https://en.wikipedia.org/wiki/Galois/Counter_Mode) with a 16-byte Initialization Vector (IV).
+
+If subsequent versions support additional encryption algorithms, those details should be exchanged between the sender and receiver when they establish the pre-shared key.
+
+### Condition and Fulfillment Generation
+
+The sender and receiver generate the condition and fulfillment from the pre-shared key that gives this protocol its name and the data that is sent with each payment. The fulfillment is an HMAC (using SHA256) of a key derived from the shared secret and the encrypted data.
+
+As in PSKv1, this method for generating the condition and fulfillment enables many payments to be sent without end-to-end communication for each one. However, because the sender also knows the fulfillment up-front, neither version of PSK allows the sender to use the fulfillment as a non-repudiable proof of payment. For use cases where this is desired, a Transport Protocol like the [Interledger Payment Request (IPR)](./0011-interledger-payment-request/0011-interledger-payment-request.md) should be used instead.
+
+### End-To-End Quoting
+
+In contrast to PSKv1, which depends on the [Interledger Quoting Protocol (ILQP)](./0008-interledger-quoting-protocol/0008-interledger-quoting-protocol.md), PSKv2 has two built-in mechanisms for discovering the exchange rate of a given path: informational quotes and dynamic price discovery.
+
+For purely informational quotes, PSKv2 uses unfulfillable test payments to the receiver to determine the exchange rate for a given path. The condition is set to a random 32-byte condition (with no known fulfillment). When the receiver gets the incoming payment, they reject the incoming transfer and include in their response how much arrived in that transfer. The sender can use the source amount they set and the receiver's response to determine the rate.
+
+When a sender is sending many payments to the receiver, they need a dynamic view of the exchange rate because the rate may change during the course of the interaction. To handle these cases, every PSKv2 payment response includes the amount that arrived in the final transfer to the receiver. Similar to the informational quotes, the sender can use the chunk source amount and the receiver's response to determine and monitor the exchange rate.
+
+**Trusting the Receiver:** With any type of Interledger quoting, whether end-to-end or using ILQP, senders SHOULD judge prices in their source units. Receivers can lie about how much arrives in a test payment for a quote or they could run a connector that happens to have a bad exchange rate to their destination units. Therefore, senders must assume that whatever amount the receiver says has arrived represents the "real" rate of the path.
```

Comment:
Strictly speaking, in ILQP the receiver cannot lie about how much arrives, unless they run the destination ledger

Reply:
"or they could run a connector that happens to have a bad exchange rate to their destination units"

Reply:
running one such connector wouldn't make a difference, the network would route around it, using other connectors to reach the destination ledger